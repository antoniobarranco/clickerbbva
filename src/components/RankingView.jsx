import React from 'react';
import PropTypes from 'prop-types';

function RankingView({ showLimit }) {
  return (
    <>
      <div>RankingView</div>
      <p>To be done...</p>
      <p>
        Showing
        {' '}
        {showLimit}
      </p>
    </>
  );
}

RankingView.propTypes = {
  showLimit: PropTypes.number.isRequired,
};

export default RankingView;
