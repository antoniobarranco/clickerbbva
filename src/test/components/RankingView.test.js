import renderer from 'react-test-renderer';
import RankingView from "../../components/RankingView";

import { MemoryRouter } from "react-router-dom";

describe('Tests on RankingView component', () => {

    test('Test RankingView with Snapshot test for UI changes', () => {

      const tree = renderer.create(
        <MemoryRouter initialEntries={['/dashboard']}>
          <RankingView />
        </MemoryRouter>,).toJSON();

      expect(tree).toMatchSnapshot();

    })

})