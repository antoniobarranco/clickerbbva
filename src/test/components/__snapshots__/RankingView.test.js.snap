// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on RankingView component Test RankingView with Snapshot test for UI changes 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-primary bg-opacity-75"
  >
    <div
      className="container-fluid"
    >
      <span
        className="navbar-brand mb-0 h1"
        data-testid="BreadCrumbsText"
      >
        Ranking
      </span>
      <div
        className="d-flex"
      >
        <button
          className="btn btn-outline-light"
          onClick={[Function]}
          type="button"
        >
          <i
            className="bi bi-box-arrow-in-right"
          />
           
          Close
        </button>
      </div>
    </div>
  </nav>,
  <div
    className="d-grid gap-2 col-8  my-5 mx-auto"
  >
    <ul
      className="list-group"
    >
      <li
        className="list-group-item d-flex justify-content-between align-items-center active"
      >
        Game Ranking
      </li>
    </ul>
  </div>,
]
`;
